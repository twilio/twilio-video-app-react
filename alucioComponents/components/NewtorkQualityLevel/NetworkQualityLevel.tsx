import React from 'react';

const Container = {
  display: 'flex',
  alignItems: 'flex-end'
};

const STEP = 3;

export default function NetworkQualityLevel({ qualityLevel }: { qualityLevel: number | null }) {
  if (qualityLevel === null) return null;
  return (
    <div style={Container}>
      {[0, 1, 2, 3, 4].map(level => (
        <div
          key={level}
          style={{
            height: `${STEP * (level + 1)}px`,
            background: qualityLevel > level ? '#0c0' : '#040',
            width: '2px',
            border: '1px solid black',
            boxSizing: 'content-box',
            borderRight: 'none'
          }
        }
        />
      ))}
    </div>
  );
}
