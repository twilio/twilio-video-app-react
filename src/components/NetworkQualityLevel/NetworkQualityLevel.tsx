import { makeStyles } from '@material-ui/core/styles';
import { Participant } from 'twilio-video';
import useParticipantNetworkQualityLevel from '../../hooks/useParticipantNetworkQualityLevel/useParticipantNetworkQualityLevel';

const useStyles = makeStyles({
  outerContainer: {
    width: '2em',
    height: '2em',
    padding: '0.9em',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: 'rgba(0, 0, 0, 0.5)',
  },
  innerContainer: {
    display: 'flex',
    alignItems: 'flex-end',
    '& div': {
      width: '2px',
      marginRight: '1px',
      '&:not(:last-child)': {
        borderRight: 'none',
      },
    },
  },
});

const STEP = 3;
const BARS_ARRAY = [0, 1, 2, 3, 4];

export default function NetworkQualityLevel({ participant }: { participant: Participant }) {
  const classes = useStyles();
  const networkQualityLevel = useParticipantNetworkQualityLevel(participant);

  if (networkQualityLevel === null) return null;

  return (
    <div className={classes.outerContainer}>
      <div className={classes.innerContainer}>
        {BARS_ARRAY.map(level => (
          <div
            key={level}
            style={{
              height: `${STEP * (level + 1)}px`,
              background: networkQualityLevel > level ? 'white' : 'rgba(255, 255, 255, 0.2)',
            }}
          />
        ))}
      </div>
    </div>
  );
}
