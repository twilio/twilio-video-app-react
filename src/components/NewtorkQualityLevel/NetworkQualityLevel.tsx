import React from 'react';
import { styled } from '@material-ui/core/styles';

const Container = styled('div')({
  display: 'flex',
  alignItems: 'flex-end',
  '& div': {
    width: '2px',
    border: '1px solid black',
    boxSizing: 'content-box',
    '&:not(:last-child)': {
      borderRight: 'none',
    },
  },
});

const STEP = 3;

export default function NetworkQualityLevel({ qualityLevel }: { qualityLevel: number | null }) {
  if (qualityLevel === null) return null;
  return (
    <Container>
      {[0, 1, 2, 3, 4].map(level => (
        <div
          key={level}
          style={{
            height: `${STEP * (level + 1)}px`,
            background: qualityLevel > level ? '#0c0' : '#040',
          }}
        />
      ))}
    </Container>
  );
}
