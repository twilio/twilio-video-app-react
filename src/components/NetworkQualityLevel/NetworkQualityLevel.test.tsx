import renderer from 'react-test-renderer';
import NetworkQualityLevel from './NetworkQualityLevel';

describe('the NetworkQualityLevel component', () => {
  it('should render correctly for level 5', () => {
    const mockParticipant = { networkQualityLevel: 5, on: () => {} } as any;
    const tree = renderer.create(<NetworkQualityLevel participant={mockParticipant} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for level 3', () => {
    const mockParticipant = { networkQualityLevel: 3, on: () => {} } as any;
    const tree = renderer.create(<NetworkQualityLevel participant={mockParticipant} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for level 0', () => {
    const mockParticipant = { networkQualityLevel: 0, on: () => {} } as any;
    const tree = renderer.create(<NetworkQualityLevel participant={mockParticipant} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
