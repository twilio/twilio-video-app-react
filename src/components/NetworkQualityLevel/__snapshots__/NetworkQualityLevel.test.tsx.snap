// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NetworkQualityLevel component should render correctly for level 0 1`] = `
<div
  className="makeStyles-outerContainer-1"
>
  <div
    className="makeStyles-innerContainer-2"
  >
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "3px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "6px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "9px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "12px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "15px",
        }
      }
    />
  </div>
</div>
`;

exports[`the NetworkQualityLevel component should render correctly for level 3 1`] = `
<div
  className="makeStyles-outerContainer-1"
>
  <div
    className="makeStyles-innerContainer-2"
  >
    <div
      style={
        Object {
          "background": "white",
          "height": "3px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "6px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "9px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "12px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "rgba(255, 255, 255, 0.2)",
          "height": "15px",
        }
      }
    />
  </div>
</div>
`;

exports[`the NetworkQualityLevel component should render correctly for level 5 1`] = `
<div
  className="makeStyles-outerContainer-1"
>
  <div
    className="makeStyles-innerContainer-2"
  >
    <div
      style={
        Object {
          "background": "white",
          "height": "3px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "6px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "9px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "12px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "white",
          "height": "15px",
        }
      }
    />
  </div>
</div>
`;
